src/Evolution/Systems/RadiationTransport/M1Grey/M1Closure.cpp

```diff
@@ -0,0 +1,284 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/RadiationTransport/M1Grey/M1Closure.hpp"
+
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <utility>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "NumericalAlgorithms/RootFinding/NewtonRaphson.hpp"
+#include "PointwiseFunctions/GeneralRelativity/IndexManipulation.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace {
+/// \cond
+struct MomentumUp {
+  using type = tnsr::I<DataVector, 3, Frame::Inertial>;
+};
+struct MomentumSquared {
+  using type = Scalar<DataVector>;
+};
+/// \endcond
+
+/// Minerbo (maximum entropy) closure for the M1 scheme
+double minerbo_closure_function(const double zeta) noexcept {
+  return 1.0 / 3.0 +
+         square(zeta) * (6.0 - 2.0 * zeta + 6.0 * square(zeta)) / 15.0;
+}
+double minerbo_closure_deriv(const double zeta) noexcept {
+  return 6. / 15 * zeta * (2. - zeta + 4. * square(zeta));
+}
+}  // namespace
+
+namespace RadiationTransport {
+namespace M1Grey {
+
+void M1Closure(
+    gsl::not_null<Scalar<DataVector>*> closure_factor,
```

Comment:
const the not_nulls (this makes a const pointer to a non-const object)

Reply:
I don't think that I understand the issue - I was following the same setup as in GrMhd/PrimitiveFromConservative.cpp (for the apply function). Should I have

const gsl::not_null<Scalar<DataVector>*> closure_factor

(and so on for the other options)? And if yes, why do we do it here but not in GrMhd?

Reply:
yes, it should be done in both places